root/admin/attributes/AttributeEditForm.js

```diff
@@ -18,40 +18,46 @@ import ScriptEditForm from './ScriptEditForm.js';
 import type {CreateOrEditAttributePropsT} from './types.js';
 import WorkAttributeTypeEditForm from './WorkAttributeTypeEditForm.js';
 
-const AttributeEditForm = ({
-  action,
-  entityTypeSelectOptions,
-  form,
-  parentSelectOptions,
-  type,
-}: CreateOrEditAttributePropsT): React.MixedElement => (
-  type === 'Language' ? (
-    <LanguageEditForm form={form} />
-  ) : type === 'Script' ? (
-    <ScriptEditForm form={form} />
-  ) : type === 'CollectionType' || type === 'SeriesType' ? (
-    <AttributeEditFormWithEntityType
-      action={action}
-      entityTypeSelectOptions={entityTypeSelectOptions}
-      form={form}
-      parentSelectOptions={parentSelectOptions}
-    />
-  ) : type === 'MediumFormat' ? (
-    <MediumFormatEditForm
-      form={form}
-      parentSelectOptions={parentSelectOptions}
-    />
-  ) : type === 'WorkAttributeType' ? (
-    <WorkAttributeTypeEditForm
-      form={form}
-      parentSelectOptions={parentSelectOptions}
-    />
-  ) : (
-    <AttributeEditGenericForm
-      form={form}
-      parentSelectOptions={parentSelectOptions}
-    />
-  )
-);
+const AttributeEditForm = (
+  props: CreateOrEditAttributePropsT,
```

Comment:
Does this work just because Flow doesn't check these as carefully as the previous way? Or is this a documented way to deal with this issue? :/

Reply:
It's documented at https://flow.org/en/docs/types/unions/#toc-disjoint-object-unions

In order to refine an object type in this way, you have to check the property on the object (which only refines the variable holding the object).  If you destructure the object into separate variables first, Flow doesn't track that those variables have any logical relation to each other.